question3/hw6q3.c

```diff
@@ -0,0 +1,295 @@
+#include<stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h>
+
+
+#define MAXLINES 5000
+#define ALLOCSIZE 10000 // size of available space
+
+// globals
+char *lineptr[MAXLINES]; // pointers to text lines
+static char allocbuf[ALLOCSIZE]; // storage for alloc
+static char *allocp = allocbuf;  // next free position
+
+// functions
+double my_atof(char s[]);
+int numcmp(char *s1, char *s2);
+int fldcmp(char *s1, char *s2);
+int revstrcmp(char *s1, char *s2);
+int revnumcmp(char *s1, char *s2);
+int revfldcmp(char *s1, char *s2);
+int revfldnumcmp(char *s1, char *s2);
+int fldnumcmp(char *s1, char *s2);
+int readlines(char *lineptr[], int nlines);
+void writelines(char *lineptr[], int nlines);
+void q_sort(void *lineptr[], int left, int right, 
+           int (*comp)(void *, void *));
+
+// sort input lines
+int main(int argc, char *argv[])
+{
+    int nlines, i; // number of input lines read
+    int numeric = 0, reverse = 0, fold = 0;
+
+    for (i = 0; i < argc; ++i)
+    {
+        if (strcmp(argv[i], "-n") == 0)
+            numeric = 1;
+        if (strcmp(argv[i], "-r") == 0)
+            reverse = 1;
+        if (strcmp(argv[i], "-f") == 0)
+            fold = 1;
+    }
+    printf("n: %d, r: %d, f: %d\n\n", numeric, reverse, fold);
+    if ((nlines = readlines(lineptr, MAXLINES)) >= 0)
+    {
+        if (numeric == 0 && reverse == 0 && fold == 0)
+            q_sort((void **) lineptr, 0, nlines-1,
+              (int (*)(void*, void*))(strcmp));
```

Comment:
nice that you used the pointer to a function! Though you could have simply used the string sort. :)